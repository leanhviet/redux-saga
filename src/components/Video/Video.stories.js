// Libs
import React from 'react'
import { storiesOf } from '@storybook/react'
import { MemoryRouter } from 'react-router-dom'

// Components
import Video from './index'

const mockData = {
  img: '',
  timeout: '05:09',
  title: 'Hope',
  publishedTime: '1 day ago',
  channelName: 'Le Van A',
  numberOfViews: '1,000,500'
}

storiesOf('Video', module)
  .add('Default', () => (
    <div className="container">
      <div className="row">
        <div className="col-md-4">
          <MemoryRouter>
            <Video />
          </MemoryRouter>
        </div>
      </div>
    </div>
  ))
  .add('Contain Data', () => (
    <div className="container">
      <div className="row">
        <div className="col-md-4">
          <MemoryRouter>
            <Video video={mockData} />
          </MemoryRouter>
        </div>
      </div>
    </div>
  ))
