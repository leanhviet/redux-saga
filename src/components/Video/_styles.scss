.video {
  margin-top: 40px;

  &__area {
    position: relative;

    &:hover &__mask {
      @include transition(all 0.3s ease-in-out);

      display: block;
      cursor: pointer;
      opacity: 100;
    }

    &__mask {
      position: absolute;
      background-color: $persimmon;
      width: 100%;
      height: 100%;
      text-align: center;
      opacity: 0;

      & .vertical-align {
        @include transform(translateY(-50%));

        position: relative;
        top: 50%;

        & i {
          font-size: 80px;
          color: $white;
        }
      }
    }

    & .duration {
      background-color: $red;
      color: $white;
      padding: 10px 30px;
      text-transform: uppercase;
      font-weight: 700;
      font-size: 12px;
      position: absolute;
      right: 0;
      bottom: 0;
    }
  }

  &__published-text {
    & > span {
      color: $red;
    }

    & > strong {
      color: $chathams-blue;
    }
  }

  & > p {
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  }
}
