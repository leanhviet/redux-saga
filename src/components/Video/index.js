// @flow
import React from 'react'
import { Link } from 'react-router-dom'

// Constants
import ROUTES from '../../constants/routes'

type Props = {
  video: Object
}

const Video = (props: Props) => {
  const {
    id,
    img,
    timeout,
    title,
    publishedTime,
    channelName,
    numberOfViews
  } = props.video
  const defaultSrc = ev =>
    (ev.target.src = 'https://via.placeholder.com/365x320')

  return (
    <div className="video">
      <div className="video__area">
        <Link to={`${ROUTES.DETAIL}/${id}`}>
          <div className="video__area__mask">
            <div className="vertical-align">
              <i className="fa fa-play-circle" />
              <p>Play Now</p>
            </div>
          </div>
          <img
            src={img}
            onError={defaultSrc}
            alt="video"
            className="img-responsive"
          />
          <span className="duration">{timeout}</span>
        </Link>
      </div>
      <Link to={`${ROUTES.DETAIL}/${id}`}>
        <h4>{title}</h4>
      </Link>
      <p className="video__published-text">
        <span>{publishedTime}</span> By <strong>{channelName}</strong>
      </p>
      <p className="views">{`${numberOfViews} Views`}</p>
    </div>
  )
}

Video.defaultProps = {
  video: {
    id: '',
    img: '',
    timeout: '00:00',
    title: '',
    publishedTime: '',
    channelName: '',
    numberOfViews: 0
  }
}

export default Video
