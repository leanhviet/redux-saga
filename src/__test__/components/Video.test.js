// Libs
import React from 'react'
import { shallowToJson } from 'enzyme-to-json'
import { shallow } from 'enzyme'

// Themes
import ImageTheme from '../../themes/images'

// Component
import Video from '../../components/Video'

const mockProps = {
  img: ImageTheme.profile,
  timeout: '05:30',
  title: 'I hope',
  publishedTime: 'Published on 2 weeks ago',
  channelName: 'Le Anh B',
  numberOfViews: '2,111,242'
}

const wrapper = (props = {}) => {
  const component = shallow(<Video video={props} />)

  return component
}

describe('Video component', () => {
  const wrapperWithoutProps = wrapper()

  // Snapshot
  test('should render correctly Video component default props', () => {
    expect(shallowToJson(wrapperWithoutProps)).toMatchSnapshot()
  })
})
