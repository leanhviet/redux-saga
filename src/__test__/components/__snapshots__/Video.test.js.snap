// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video component should render correctly Video component default props 1`] = `
<div
  className="video"
>
  <div
    className="video__area"
  >
    <Link
      to="/detail/undefined"
    >
      <div
        className="video__area__mask"
      >
        <div
          className="vertical-align"
        >
          <i
            className="fa fa-play-circle"
          />
          <p>
            Play Now
          </p>
        </div>
      </div>
      <img
        alt="video"
        className="img-responsive"
        onError={[Function]}
      />
      <span
        className="duration"
      />
    </Link>
  </div>
  <Link
    to="/detail/undefined"
  >
    <h4 />
  </Link>
  <p
    className="video__published-text"
  >
    <span />
     By 
    <strong />
  </p>
  <p
    className="views"
  >
    undefined Views
  </p>
</div>
`;
